tests/framework/exec_util.go

```diff
@@ -0,0 +1,143 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package framework
+
+import (
+	"bytes"
+	"io"
+	"net/url"
+	"strings"
+
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/scheme"
+	restclient "k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/remotecommand"
+
+	. "github.com/onsi/gomega"
+)
+
+// ExecOptions passed to ExecWithOptions
+type ExecOptions struct {
+	Command []string
+
+	Namespace     string
```

Comment:
name + ns are fine. Did you consider passing *v1.Pod?

Reply:
No since I copied and pasted this from the k8s e2e framework. Seems like that part of the code is somewhat old and could use some improvements.

Reply:
@awels Yes, that is one of the challenges of copying code -- you're never sure if what you're grabbing is actually the preferred way of doing something. K8s e2e is a good example of both old and current implementations of e2e testing, and I do not always know which is which either. What you have above is fine. I was just thinking that having a pointer to a pod might be better because 1) it's just one field, 2) you likely have already created this pod, and 3) maybe future enhancement to these functions will be need other pod fields in the exec options.

Reply:
@jeffvance In general I prefer structs over a bunch of strings as it makes it much clearer what we are talking about. In this particular case I am on the fence. As you say its highly likely someone will have the pod already and passing it in is easy. On the other hand we will internally simply do pod.GetName and pod.GetNamespace. And if someone didn't have the pod, they will have to construct it from the name and namespace, only for us to just grab those two fields.

So in the case that the caller has the pod already, its easy for them to pass the two fields we are interested in. And in the case they don't its also easy for them to pass the fields as they have them already. And they don't have to construct the Pod struct at all.

Reply:
@awels I'm ok leaving it the way you originally wrote it. If there's a need later it can easily be changed.